test_communication/test/test_subscriber.cpp

```diff
@@ -57,8 +57,8 @@ rclcpp::subscription::SubscriptionBase::SharedPtr subscribe(
       }
 
       // shutdown node when all expected messages have been received
-      for (auto received : received_messages) {
-        if (!received) {
+      for (auto received_msg : received_messages) {
+        if (!received_msg) {
```

Comment:
Actually I think that was not a pedantic warning but a windows warning when I was trying more permissive flags. I think it's because it is shadowing the variable of the same name defined [line 39](https://github.com/ros2/system_tests/blob/087c3743e06eae7b1d82c175835a82d15fe31ab2/test_communication/test/test_subscriber.cpp#L39)